### PR TITLE
collector extension initial commit

### PR DESCRIPTION
**Pull request by wangzlei**
* create strip down collector components, open for adding useful components.
* upgrade to the latest Collector 0.22.0
* by default use otlp reciever and logging exporter

### CODE REVIEW

File: collector/Makefile [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+BUCKET_NAME:=lambda-artifacts-$(shell dd if=/dev/random bs=8 count=1 2>/dev/null | od -An -tx1 | tr -d ' \t\n')
+LAYER_NAME:=otel-collector
+
+VERSION=$(shell cat VERSION)
+GIT_SHA=$(shell git rev-parse HEAD)
+GOBUILD=GO111MODULE=on CGO_ENABLED=0 installsuffix=cgo go build -trimpath
+BUILD_INFO_IMPORT_PATH=main
+
+LDFLAGS=-ldflags "-s -w -X $(BUILD_INFO_IMPORT_PATH).GitHash=$(GIT_SHA) -X $(BUILD_INFO_IMPORT_PATH).Version=$(VERSION) \
+-X github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awsxrayexporter.collectorDistribution=aws-otel-collector-lambda"
```
Comment (anuraaga) [inline collector/Makefile]:
```suggestion
-X github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awsxrayexporter.collectorDistribution=opentelemetry-collector-lambda"
```

File: collector/defaultcomponents/default.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+		 errs = append(errs, err)
+	 }
+ 
+	 factories := component.Factories{
+		 Receivers:  receivers,
+		 Exporters:  exporters,
+	 }
+ 
+	 return factories, consumererror.CombineErrors(errs)
+ }
```
Comment (anuraaga) [inline collector/defaultcomponents/default.go]: gofmt

File: collector/README.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
+# OpenTelemetry Collector AWS Lambda Extension layer
+
+The OpenTelemetry Collector Lambda Extension provides a mechanism to export telemetry aynchronously from AWS Lambdas. It does this by embedding an [AWS Distro for OpenTelemetry Collector(AWS OTel Collector)](https://github.com/aws-observability/aws-otel-collector) inside an [AWS Extension Layer](https://aws.amazon.com/blogs/compute/introducing-aws-lambda-extensions-in-preview/). This allows lambdas to use the OpenTelemetry Collector Exporter to send traces and metrics to any configured backend.
```
Comment (anuraaga) [inline collector/README.md]: Update now it's OpenTelemetry Collector Contrib

File: collector/README.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+      exporters: [logging, otlp]
+```
+
+Once the file has been deployed with a Lambda, configuring the `OPENTELEMETRY_COLLECTOR_CONFIG_FILE` will tell the OpenTelemetry extension where to find the collector configuration:
+
+```
+aws lambda update-function-configuration --function-name Function --environment Variables={OPENTELEMETRY_COLLECTOR_CONFIG_FILE=/var/task/collector.yaml}
+```
+
+You can configure environment variables via yaml as well:
```
Comment (anuraaga) [inline collector/README.md]:
```suggestion
You can configure environment variables via SAM as well:
```

File: collector/README.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  Function:
+    Type: AWS::Serverless::Function
+    Properties:
+      ...
+      Environment:
+        Variables:
+          OPENTELEMETRY_COLLECTOR_CONFIG_FILE: /var/task/collector.yaml
+```
+
+## License
```
Comment (anuraaga) [inline collector/README.md]: We don't need this in each subdirectory

File: collector/go.mod [hunk 6]
```diff
@@ -0,0 +1,14 @@
+module github.com/open-telemetry/opentelemetry-lambda/collector
+
+go 1.14
+
+require (
+	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awsemfexporter v0.22.0
```
Comment (anuraaga) [inline collector/go.mod:6]: Thanks

File: collector/go.mod [hunk 7]
```diff
@@ -0,0 +1,14 @@
+module github.com/open-telemetry/opentelemetry-lambda/collector
+
+go 1.14
```
Comment (anuraaga) [inline collector/go.mod]:
```suggestion
go 1.16
```

New project so may as well

File: collector/README.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+      exporters: [logging, otlp]
+```
+
+Once the file has been deployed with a Lambda, configuring the `OPENTELEMETRY_COLLECTOR_CONFIG_FILE` will tell the OpenTelemetry extension where to find the collector configuration:
+
+```
+aws lambda update-function-configuration --function-name Function --environment Variables={OPENTELEMETRY_COLLECTOR_CONFIG_FILE=/var/task/collector.yaml}
+```
+
+You can configure environment variables via cloudformation template as well:
```
Comment (anuraaga) [inline collector/README.md]: Why do we say cloudformation template here and SAM above?

Comment (wangzlei) [reply collector/README.md]: yeah, I think the correct way should be CloudFormation template but not SAM, will update the above place.

**Review by anuraaga — APPROVED** [review]

**Pull request merged**